scripts/demo_common.sh

```diff
@@ -16,14 +16,32 @@
 
 set -e -u
 
-## $root_dir is the root directory of the Mbed TLS source tree.
+is_mbedtls_root () {
+  (
+    cd $1
+    if in_mbedtls_repo; then
+      return 1;
+    else
+      return 0;
+    fi
+  )
+}
+
+## At the end of the while loop below $root_dir will point to the root directory
+## of either the Mbed TLS/TF-PSA-Crypto source tree (the one that is found first).
 root_dir="${0%/*}"
-# Find a nice path to the root directory, avoiding unnecessary "../".
-# The code supports demo scripts nested up to 4 levels deep.
-# The code works no matter where the demo script is relative to the current
-# directory, even if it is called with a relative path.
+## Find a nice path to the root directory, avoiding unnecessary "../".
+##
+## The code supports demo scripts nested up to 4 levels deep.
+##
+## The code works no matter where the demo script is relative to the current
+## directory, even if it is called with a relative path.
+##
+## This search is based on the fact that both Mbed TLS and TF-PSA-Crypto has
+## a file in "<root>/scripts/project_name.txt" which contains the name of the
+## project. We take advantage of this file to check if we reached the right path.
 n=4 # limit the search depth
-while ! [ -d "$root_dir/programs" ] || ! [ -d "$root_dir/library" ]; do
+while ! [ -f "$root_dir/scripts/project_name.txt" ]; do
```

Comment:
This breaks `tf-psa-crypto/programs/psa/key_ladder_demo.sh` and `tf-psa-crypto/programs/psa/key_ladder_demo.sh` and thus `tests/scripts/run_demos.py` as `root_dir` then corresponds to `tf-psa-crypto` and there is no `query_compile_time_config` in TF-PSA-Crypto currently. What about to first search for mbedtls root and if we do not find it, search for TF-PSA-Crypto root?